app/views/rails_admin/main/_show_hide_js.html.haml

```diff
@@ -1,11 +1,38 @@
 :javascript
   $(document).ready(function() {
-    document.getElementById('game_prizes_text_field').style.display = "none";
-    document.getElementById('game_completion_certificate_template_field').style.display = "none";
+    // Hide toggleable fields (Game)
+    try {
+      document.getElementById('game_prizes_text_field').style.display = "none";
+      document.getElementById('game_completion_certificate_template_field').style.display = "none";
+    } catch {}
 
-    document.getElementById('game_enable_completion_certificates').addEventListener('change', toggleDisplayCert);
-    document.getElementById('game_prizes_available').addEventListener('change', toggleDisplayPrizes);
+    // Toggle sponsor fields based on if the checkmark is checked
+    try {
+      toggleSponsorField()
+    } catch {}
+    
+    // Listen for completion certificates and prizes tickbox
+    try {
+      document.getElementById('game_enable_completion_certificates').addEventListener('change', toggleDisplayCert);
+      document.getElementById('game_prizes_available').addEventListener('change', toggleDisplayPrizes);
+    } catch {}
 
+    // Listen for sponsorship tickbox
+    try {
+      $('.sponsorship_fields_toggler').find('input')[1].addEventListener('change', toggleSponsorField);
+    } catch {}
+
+    // Show or hide sponsorship info based on sponsored tickbox
+    function toggleSponsorField() {
+      var elems = $('.sponsorship_fields_toggle')
+      var arr = jQuery.makeArray( elems );
+      if ($('.sponsorship_fields_toggler').find('input')[1].checked) {
+        arr.forEach(element => element.style.display = "");
```

Comment:
It shouldn't be necessary to use `forEach` as part of this, you should just be able to say 

```javascript
var elems = $('.sponsorship_fields_toggle')
elems.hide() // or elems.show() or elems.toggle()
```